Sources/ParseSwift/API/API+Commands.swift

```diff
@@ -55,6 +55,62 @@ internal extension API {
             }
         }
 
+        public func executeAsync(options: API.Options, completion: @escaping(U?, ParseError?) -> Void) {
+            let params = self.params?.getQueryItems()
+            let headers = API.getHeaders(options: options)
+            let url = ParseConfiguration.serverURL.appendingPathComponent(path.urlComponent)
+
+            var components = URLComponents(url: url, resolvingAgainstBaseURL: false)!
+            components.queryItems = params
+
+            var urlRequest = URLRequest(url: components.url!)
+            urlRequest.allHTTPHeaderFields = headers
+            if let body = data {
+                urlRequest.httpBody = body
+            }
+            urlRequest.httpMethod = method.rawValue
+
+            if #available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.0, *) {
+                _ = URLSession.shared.asyncDataTask(with: urlRequest)
+                    .sink(receiveCompletion: { errorCompletion in
+                        if case let .failure(error) = errorCompletion {
+                            completion(nil, error)
+                        }
+                    }, receiveValue: { responseData in
+
+                        guard let decoded = try? self.mapper(responseData) else {
+                            guard let parseError = try? getDecoder().decode(ParseError.self, from: responseData) else {
+                                completion(nil, ParseError(code: .unknownError, message: "cannot decode error"))
+                                return
+                            }
+                            completion(nil, parseError)
+                            return
+                        }
+                        completion(decoded, nil)
+                    })
```

Comment:
With combine, you typically would add the subscription to the publisher here. If it was an instance that wasn’t killed, it would be easy and solve the problem. It’s not as easy here because it’s a value type and can’t be mutated.